Source/WebCore/html/HTMLCanvasElement.cpp

```diff
@@ -766,7 +766,8 @@ RefPtr<ImageData> HTMLCanvasElement::getImageData()
     if (is<WebGLRenderingContextBase>(m_context)) {
         if (RuntimeEnabledFeatures::sharedFeatures().webAPIStatisticsEnabled())
             ResourceLoadObserver::shared().logCanvasRead(document());
-        return ImageData::create(downcast<WebGLRenderingContextBase>(*m_context).paintRenderingResultsToPixelBuffer());
+        auto pixelBuffer = downcast<WebGLRenderingContextBase>(*m_context).paintRenderingResultsToPixelBuffer();
+        return ImageData::create(static_pointer_cast<Uint8ClampedPixelBuffer>(WTFMove(pixelBuffer)));
```

Comment:
I guess we know that WebGLRenderingContextBase will return a Uint8ClampedPixelBuffer?

Reply:
Yes it should be of type ByteArrayPixelBuffer because GraphicsContextGLANGLE::readPixelsForPaintResults() calls ByteArrayPixelBuffer::tryCreate() for the returned PixelBuffer. But I added the new if-statement before this casting 
```
if (!is<ByteArrayPixelBuffer>(pixelBuffer))
    return nullptr;
```
to make it clear that this is the expectation.